docs/adr/0005-graduation of EventBridge.md

```diff
@@ -0,0 +1,57 @@
+# 5. Graduation of EventBridge in CAPA
+
+* Status: [proposed]
+* Date: TBD 
+* Authors: @Ankitasw
+* Deciders: @richardcase @sedefsavas
+
+## Context
+EventBridge is an eventbus provided by AWS to watch for the events generated by any SaaS application running on AWS, or any of the AWS services, that could be used by any other services.
+
+With respect to CAPA, EventBridge helps watch EC2 instance state change events using which _AWSMachine_ controller would trigger reconciliation and handle the lifecycle of the EC2 instances.
+
+There are other `CloudWatch` events that exists which can be watched upon and CAPA controllers can take necessary actions based on the events, for eg. ASG Scale-in events, Spot Instance Termination Notices, Scheduled Maintenance events etc.
+
+## Decision
+This document describes the ways on how to accommodate watching different types of events and how controllers could utilize these events. The scenarios that would need modification in EventBridge in CAPA (in `AWSInstanceState` controller) are as below:
+* In the current _AWSMachinePool_ implementation, we don't have node draining logic. With the use of lifecycle hooks, CAPA controllers need to watch `ASG scale-in` events and act on them such that the EC2 instances shuts down gracefully.
+* We also plan to support spot instances with _AWSMachinePool_ which would require CAPA controllers to watch for `Spot Instance Termination Notices(SITN)` which would give enough time to drain spot instances gracefully before they shut down.
+* Handle scheduled maintenance events where the EC2 instance needs to be restarted because of reasons like hardware/software updates.
+
+The `AWSInstanceState` controller would accommodate different types of events and would be flexible enough to be used across CAPA controllers.
+
+## Consequences
+### Current Design
```

Comment:
I don't we should be covering "current design" in the consequences section. We really only need to cover the what will change as a result of this decision....so the stuff in the "proposed design" section.

Reply:
Shall we add this in decision section, or create a separate section?

Reply:
I think there is too much detail overall. I think the decision section only needs to be a few sentances max.

Reply:
Since this is a bigger change, I wanted to highlight design decision as well, so if I want to add the design part, can I add a separate section in ADR only? Otherwise if I directly raise PR, it would be too huge to understand. But since this was not that bigger feature change, I thought of adding this to ADR, wdyt @richardcase ?

Reply:
or else i create a separate google doc and just reference it in decision section.

Reply:
In that case, yes let's add the detail in the "current design" to the context.

Reply:
and proposed design in decision section only?

Reply:
Proposed design in the consequences. Keep the decision small.

Reply:
sure, thanks for the inputs @richardcase 🙂